### DISCUSSION TITLE
srsRAN containers form factor

### DISCUSSION THREAD

**Discussion by jmarkotic** [srsran/srsRAN_Project#185] [General Help] [answered]
Hi,
i am quite new to srsRAN, learning now. Thank you for great work done.
I was wndering if there is srsRAN containers form factor that can be downloaded. and installed on kubernetes cluster ?
i found tha redhat offers those (CU, DU containers) on their repository ? I could not find it on srsRAN site.

**Comment by ninjab3s** [comment, marked as answer]
Hi @jmarkotic,
we are currently working on something there. There is a container on docker.io which you can download (https://hub.docker.com/repository/docker/softwareradiosystems/srsran-cu-du-ubi8/general). This is the repo we use for pushing our containers to RedHat, you can download them from there. Unfortunately, the container is still under development. Expect it to work in about 2/3 weeks. We are also working on a helm chart and an operator for Kubernetes. Those are not yet available but will also be in approx. 2/3 weeks.

**Comment by jmarkotic** [comment]
thanks @ninjab3s . I will monitor for initial helm charts.

**Comment by joshuamoorexyz** [comment]
Any update on this? With this would the near rt ric and xapps as described in the documentation be compatible allowing full containerization?

**Reply by pgawlowicz** [reply]
No, the near-rt RIC and xapps described in the app note are from FlexRIC framework that is a third-party project.

**Reply by rolandmartinson7890** [reply]
@pgawlowicz So in Ubuntu, will we be able (when the update is released) to download the DU and CU into separate Ubuntu nodes and link multiple DU addresses to one CU?

**Reply by ninjab3s** [reply]
The CU/DU will remain as one binary for the next releases and therefore, it will be only one container. You cannot deploy CU and DU on different machines.

@joshuamoorexyz regarding your question, the Helm chart is not ready yet. We will release it in two weeks prior to a workshop we hold. We will announce this here, maybe on other channels as well.

**Reply by pgawlowicz** [reply]
I misunderstood your question. Each component (i.e., DU, CU) is (or will be) equipped with its own E2 agent, so each can talk to Near-RT RIC. 

Regarding, the second question regarding the split of CU and DU. We are planning to do it, but probably not in the next of months.
